abap-code-review/src/scenarios.adoc

```diff
@@ -0,0 +1,54 @@
+== Scenarios
+
+In this section we look at typical scenarios how git-based code reviews and classical ABAP CTS transports can be combined.
+We assume the SAP ABAP development application servers have network access to the code review platform, e.g., GitHub.
+
+=== One Way Synchronization
+
+By one-way we mean that CTS is the primary source of truth of your component. It is in full control of your software lifecycle. Git is only used for code review purposes as a secondary persistence or USB-stick-like export channel.
+Developers work normally, releasing transports via CTS as configured per transport layer.
+
+This setup is recommended for scnearios where multiple developers work on the same repostiory in the same system,
+pulling code into the system might overwrite ongoing changes by other developers, allowing only pushes will avoid this problem.
+
+[plantuml,one-way,svg,align="center"]
+....
+@startuml
+skinparam monochrome true
+actor developer1
+actor developer2
+actor developer3
+cloud git
+rectangle abap
+developer1 -down-> abap
+developer2 -down-> abap
+developer3 -down-> abap
+abap -> git: push
+@enduml
+....
+
+The only difference are the new review steps, which are conducted via an ABAP git client (either abapGit or gCTS) on a code review platform like GitHub.
+Even though code reviews could also be conducted independently from transport releases, it is more efficient to synchronize both tightly because their granularity typically captures a logical unit of work.
+
+This scenario requires minimal git knowledge, as developers will continue working with normal CTS requests/tasks.
+
+=== Two Way Synchronization
+
+In contrast, open source projects or reusable assets developed by partners for multiple clients usually have a lifecycle of their own, independent of the target systems they are deployed to. Git becomes the primary persistence. Classical CTS is only used optionally for software logistics of the consuming target project (or you rely completely on gCTS).
+All git operations are manually performed by the developer using the ABAP git client (either abapGit or gCTS), including the pull of changes back from git into AS ABAP.
```

Comment:
The pull to target systems can be automated.

Reply:
with code reviews in mind, the target system would be a system where development happens, I guess its typically not desired to pull automatically into these systems, as development can be ongoing?

Reply:
Sounds debatable to me. Suggest to tread this in a separate issue/PR once the initial PR is merged.